### DISCUSSION TITLE
Searching for user's repositories leads to "Unprocessable Entity" failure

### DISCUSSION THREAD

**Discussion by ilia-cy** [community/community#24572] [Apps, API and Webhooks]
<p>I’m trying to get the public repositories of a given Github user using the repository Search API.</p>
<p>For some of the users it works properly, but for some I receive the following:</p>
<pre><code>{
  "message": "Validation Failed",
  "errors": [
    {
      "message": "The listed users and repositories cannot be searched either because the resources do not exist or you do not have permission to view them.",
      "resource": "Search",
      "field": "q",
      "code": "invalid"
    }
  ],
  "documentation_url": "https://developer.github.com/v3/search/"
</code></pre>
<p>For example, the following query will trigger the error: <a href="https://api.github.com/search/repositories?q=user:ironshay" rel="nofollow noopener">https://api.github.com/search/repositories?q=user:ironshay</a></p>
<p>Going via the web UI (<a href="https://github.com/ironshay?tab=repositories" rel="nofollow noopener">https://github.com/ironshay?tab=repositories</a>) leads to a message that the user has no public repositories, which is what I expected the search API to return as well. </p>
<p>Can you please provide details regarding this error and why it happens?</p>
<p>Thanks</p>

**Comment by ilia-cy** [comment, marked as answer]
<p>For whoever is interested in this, it seems that this error can appear if you search for a user that has only private repositories and no public repositories whatsoever.</p>
<p>So as I understand, you have several response options:</p>
<ol>
<li>
<p>User has no private repositories - you will receive a response with his public repositories.</p>
</li>
<li>
<p>User has private repositories and public repositories - you will receive a response with his public repositories.</p>
</li>
<li>
<p>User has only private repositories - you will receive this “Validation failed” error.</p>
</li>
</ol>
<p><a href="https://developer.github.com/enterprise/2.20/v3/search/#access-errors-or-missing-search-results" rel="nofollow noopener">More details here</a></p>
<p>  </p>

**Reply by UmerMo** [reply]
I don't know why this is marked as an answer without a resolution? I have private repository so how can I resolve this without having to make it public.
